apps/console-lite/src/app/components/horizontal-menu/horizontal-menu.tsx

```diff
@@ -0,0 +1,103 @@
+import classNames from 'classnames';
+import React, { useEffect, useRef, useState } from 'react';
+import { Link } from 'react-router-dom';
+
+export interface Item {
+  name: string;
+  id: string;
+  url?: string;
+  isActive?: boolean;
+  color?: string;
+  cssClass?: string;
+}
+
+interface Props {
+  active?: string;
+  cssClass?: string[];
+  items: Item[];
+  'data-testid'?: string;
+  'aria-label'?: string;
+}
+
+const MenuItem = ({ id, name, url, isActive, cssClass }: Item): JSX.Element => {
+  if (!url) {
+    return <span data-testid={id}>{name}</span>;
+  }
+  return (
+    <Link
+      to={url}
+      aria-label={name}
+      className={classNames('pl-0 hover:opacity-75', cssClass, {
+        active: isActive,
+      })}
+      data-testid={id}
+    >
+      {name}
+    </Link>
+  );
+};
+
+const findActive = (active?: string, items?: Item[]) => {
+  if (!active && items?.length) {
+    return 0;
+  }
+  return items?.length
+    ? items.findIndex((item) => item.id === active) || 0
+    : -1;
+};
+
+export const HorizontalMenu = ({
+  items,
+  active,
+  'data-testid': dataTestId,
+  'aria-label': ariaLabel,
+}: Props) => {
+  const [activeNumber, setActiveNumber] = useState<number>(
+    findActive(active, items)
+  );
+  const slideContRef = useRef<HTMLUListElement | null>(null);
+  const [sliderStyles, setSliderStyles] = useState<Record<string, string>>({});
+
+  useEffect(() => {
+    setActiveNumber(findActive(active, items));
```

Comment:
All this proposals make the component less encapsulated, because need to put some logic outside. useEffect works here as counterpart of componentDidUpdate and imho it's currently the best compromise here..